core/src/main/java/com/linecorp/armeria/common/stream/InputStreamStreamMessage.java

```diff
@@ -234,6 +234,10 @@ private void request0(long n) {
             if (closed) {
                 return;
             }
+            if (offset == end) {
```

Comment:
A little bit of defensive coding
```suggestion
            if (offset >= end) {
```

Reply:
✅ updated!